Creating a STRING:

In [1]:
#creating a string
letter='p' #A string could be a single letter or bunch of texts
print(letter)
print(len(letter))
greeting='Hello World!'  #string could be made using single quote ,double quote
print(greeting)
print(len(greeting))

sentence='I hope you are enjoying 30days of python'
print(sentence)

p
1
Hello World!
12
I hope you are enjoying 30days of python


Multiline string is created by using triple single (''') or triple double quotes ("""). See the example below.



In [4]:
multiline_string='''I am Mounika Annareddy.
I didnt find anything as rewarding as empowering people.
That is why i created 30days of python'''
print(multiline_string)

multiline_string="""I am Mounika Annareddy.
I didnt find anything as rewarding as empowering people.
that is why i created 30days of python."""
print(multiline_string)

I am Mounika Annareddy.
I didnt find anything as rewarding as empowering people.
That is why i created 30days of python
I am Mounika Annareddy.
I didnt find anything as rewarding as empowering people.
that is why i created 30days of python.


String Concatenation
We can connect strings together. Merging or connecting strings is called concatenation. 

In [10]:
first_name='mounika'
last_name='Annareddy'
space=' '
full_name=first_name + space + last_name
print(full_name)

print(len(first_name))
print(len(last_name))
print((len(first_name)) > len(last_name))
print(len(full_name))

mounika Annareddy
7
9
False
17


Escape Sequences in Strings
In Python and other programming languages \ followed by a character is an escape sequence. Let us see the most common escape characters:

\n: new line
\t: Tab means(8 spaces)
\\: Back slash
\': Single quote (')
\": Double quote (")

In [13]:
print('i hope everyone is enjoying the python challenge.\n Are you?')
print('days\tTopics\tExercises')
print('day1\t5\t6')
print('day2\t6\t4')
print('day3\t7\t4')
print('day5\t7\t4')
print('This is blackslash symbol(\\)')
print('In every programming language it starts with \"hello world!\"')




i hope everyone is enjoying the python challenge.
 Are you?
days	Topics	Exercises
day1	5	6
day2	6	4
day3	7	4
day5	7	4
This is blackslash symbol(\)
In every programming language it starts with "hello world!"


String formatting
Old Style String Formatting (% Operator)
In Python there are many ways of formatting strings. In this section, we will cover some of them. The "%" operator is used to format a set of variables enclosed in a "tuple" (a fixed size list), together with a format string, which contains normal text together with "argument specifiers", special symbols like "%s", "%d", "%f", "%.number of digitsf".

%s - String (or any object with a string representation, like numbers)
%d - Integers
%f - Floating point numbers
"%.number of digitsf" - Floating point numbers with fixed precision

In [14]:
#strings only
first_name='mounika'
last_name='Annareddy'
language='python'
formatted_string=('I am %s %s.I teach %s' %(first_name,last_name,language))
print(formatted_string)

I am mounika Annareddy.I teach python


In [15]:
#strings and numbers
radius=10
pi=3.14
area=pi*radius**2
formatted_string=('the area of circle is %d is %.2f' %(radius,area))
print(formatted_string)

the area of circle is 10 is 314.00


In [18]:
python_libraries=['numpy','pandas','matplotib','flask','Django']
formatted_string=('the following are python libraries: %s' %(python_libraries))
print(formatted_string)

the following are python libraries: ['numpy', 'pandas', 'matplotib', 'flask', 'Django']


In [19]:
#New Style String Formatting (str.format)
#This formatting is introduced in Python version 3.
first_name='mounika'
last_name='Annareddy'
language='pythoon'
formatted_string=('i am {} {}.I teach {} '.format(first_name,last_name,language))
print(formatted_string)

i am mounika Annareddy.I teach pythoon 


In [26]:
a=4
b=3
print('{} + {} = {}' .format(a,b,a+b))
print('{} - {} = {}' .format(a,b,a-b))
print('{} * {} = {}' .format(a,b,a*b))
print('{} % {} = {}'.format(a,b,a%b))
print('{} / {} = {:.2f}' .format(a,b,a/b))
print('{} // {} = {}'.format(a,b,a//b))
print('{} ** {} = {}'.format(a,b,a**b))

4 + 3 = 7
4 - 3 = 1
4 * 3 = 12
4 % 3 = 1
4 / 3 = 1.33
4 // 3 = 1
4 ** 3 = 64


String Interpolation / f-Strings (Python 3.6+)
Another new string formatting is string interpolation, f-strings. Strings start with f and we can inject the data in their corresponding positions.



In [28]:
a=4
b=3
print(f'{a} + {b} = {a+b}')
print(f'{a} - {b} = {a-b}')
print(f'{a} * {b} = {a*b}')
print(f'{a} / {b} = {a/b:.2f}')
print(f'{a} // {b} = {a//b}')
print(f'{a} ** {b} = {a**b}')




4 + 3 = 7
4 - 3 = 1
4 * 3 = 12
4 / 3 = 1.33
4 // 3 = 1
4 ** 3 = 64


In [31]:
#unpacking characters
language='python'
a,b,c,d,e,f=language  # unpacking sequence characters into variables
print(a)
print(b)
print(c)
print(d)
print(e)
print(f)




p
y
t
h
o
n


Accessing Characters in Strings by Index
In programming counting starts from zero. Therefore the first letter of a string is at zero index and the last letter of a string is the length of a string minus one.

In [32]:
language='python'
first_letter=language[0]
print(first_letter)
second_letter=language[1]
print(second_letter)
third_letter=language[2]
print(third_letter)

p
y
t


In [2]:
language='python'
last_index=len(language)-1
last_letter=language[last_index]
print(last_letter)

n


In [3]:
#If we want to start from right end we can use negative indexing. -1 is the last index.
language='python'
first_letter=language[-1]
second_letter=language[-2]
print(first_letter)
print(second_letter)

n
o


Slicing Python Strings
In python we can slice strings into substrings.

In [5]:
#slicing
language='python'
first_three=language[0:3]  ## starts at zero index and up to 3 but not include 3
print(first_three)
last_three=language[3:6]
print(last_three)

#another way

first_three=language[:3]
print(first_three)
last_three=language[3:]
print(last_three)

pyt
hon
pyt
hon


Reversing a String
We can easily reverse strings in python.

In [6]:
#reversing
greeting='hello world!'
print(greeting[::-1])

!dlrow olleh


Skipping Characters While Slicing
It is possible to skip characters while slicing by passing step argument to slice method.

In [2]:
#skipping
language='python'
c=language[0:6:2]
print(c)

pto


String Methods
There are many string methods which allow us to format strings. See some of the string methods in the following example:

capitalize(): Converts the first character of the string to capital letter

In [11]:
challenge='thirty Days of Python'
print(challenge.capitalize())

Thirty days of python


count(): returns occurrences of substring in string, count(substring, start=.., end=..). The start is a starting indexing for counting and end is the last index to count.

In [12]:
challenge='thirty Days of python'
print(challenge.count('y'))
print(challenge.count('y',7,14))
print(challenge.count('th'))

3
1
2


endswith(): Checks if a string ends with a specified ending

In [14]:
challenge='thirty days of python'
print(challenge.endswith('tion'))
print(challenge.endswith('on'))

False
True


expandtabs(): Replaces tab character with spaces, default tab size is 8. It takes tab size argument

In [5]:
challenge='thirty\tdays\tof\tpython'
print(challenge.expandtabs())
print(challenge.expandtabs(10))


thirty  days    of      python
thirty    days      of        python


find(): Returns the index of the first occurrence of a substring, if not found returns -1


In [1]:
challenge='thirty days of python'
print(challenge.find('y'))
print(challenge.find('th'))

5
0


rfind(): Returns the index of the last occurrence of a substring, if not found returns -1


In [9]:
challenge='thirty days of python'
print(challenge.rfind('y'))
print(challenge.rfind('th'))

16
17


format(): formats string into a nicer output

In [12]:
first_name='Mounika'
last_name='Annareddy'
age=22
job='frelancer'
country='india'
sentence=('I am {} {}.I am {} years old.I am working as a {} . I am living in {}.'.format(first_name,last_name,age,job,country))
print(sentence)

I am Mounika Annareddy.I am 22 years old.I am working as a frelancer . I am living in india.


In [13]:
radius=10
pi=3.14
area=pi*radius**2
result=('The area of circle with radius {} is {}'.format(str(radius),str(area)))
print(result)


The area of circle with radius 10 is 314.0


index(): Returns the lowest index of a substring, additional arguments indicate starting and ending index (default 0 and string length - 1). If the substring is not found it raises a valueError.

In [19]:
challenge='thirty days of challenge'
sub_string='da'
print(challenge.index(sub_string))



7


rindex(): Returns the highest index of a substring, additional arguments indicate starting and ending index (default 0 and string length - 1)

In [25]:
challenge='thirty days of python'
sub_string='p'
print(challenge.rindex(sub_string))
print(challenge.rindex(sub_string,9))

15
15


isalnum(): Checks alphanumeric character

In [28]:
challenge='thirtydaysofpython'
print(challenge.isalnum())

True


In [29]:
challenge='30daysofpython'
print(challenge.isalnum())

True


In [30]:
challenge='thirty days of python'  #space is not an alpha numeric
print(challenge.isalnum())

False


In [31]:
challenge='thirty days of python 2019'
print(challenge.isalnum())

False


isalpha(): Checks if all string elements are alphabet characters (a-z and A-Z)


In [2]:
challenge='thirty days of python'
print(challenge.isalpha())  #space is once again excluded
challenge='thirtydayspython'
print(challenge.isalpha())
challenge='123'
print(challenge.isalpha())

False
True
False


isdecimal(): Checks if all characters in a string are decimal (0-9)

In [8]:
challenge='thirty days of python'
print(challenge.isdecimal())
challenge='123'
print(challenge.isdecimal())
challenge = '\u00B2'
print(challenge.isdecimal())
challenge='12 3'
print(challenge.isdecimal())     #space is not allowed


False
True
False
False


isdigit(): Checks if all characters in a string are numbers (0-9 and some other unicode characters for numbers)

In [9]:
challenge='thirty'
print(challenge.isdigit())
challenge='30'
print(challenge.isdigit())
challenge='\u00B2'
print(challenge.isdigit())


False
True
True


isnumeric(): Checks if all characters in a string are numbers or number related (just like isdigit(), just accepts more symbols, like ½)



In [13]:
num='10'
print(num.isnumeric())
num= '\u00BD' # ½
print(num.isnumeric())
num='10.5'
print(num.isnumeric())

True
True
False


isidentifier(): Checks for a valid identifier - it checks if a string is a valid variable name


In [38]:
challenge='30daysofpython'
print(challenge.isidentifier())    #it starts with a number
challenge='thirty_days_of_python'
print(challenge.isidentifier())

False
True


islower(): Checks if all alphabet characters in the string are lowercase

In [16]:
challenge='thirty days of python'
print(challenge.islower())
challenge='Thirty days of python'
print(challenge.islower())

True
False


isupper(): Checks if all alphabet characters in the string are uppercase

In [17]:
challenge='thirty days of python'
print(challenge.isupper())
challenge='THIRTY DAYS OF PYTHON'
print(challenge.isupper())

False
True


join(): Returns a concatenated string


In [18]:
web_tech=['html','css','javascript']
result=' '.join(web_tech)
print(web_tech)

['html', 'css', 'javascript']


strip(): Removes all given characters starting from the beginning and end of the string

In [19]:
challenge='thirty days of python'
print(challenge.strip('noth'))

irty days of py


replace(): Replaces substring with a given string

In [20]:
challenge='thirty days of python'
print(challenge.replace('python','coding'))

thirty days of coding


split(): Splits the string, using given string or space as a separator

In [29]:
challenge='thirty days of python'
print(challenge.split( ' '))


challenge='thirty,days,of,python'
print(challenge.split(' '))

['thirty', 'days', 'of', 'python']
['thirty,days,of,python']


title(): Returns a title cased string

In [30]:
challenge='thirty days of python'
print(challenge.title())

Thirty Days Of Python


swapcase(): Converts all uppercase characters to lowercase and all lowercase characters to uppercase characters


In [31]:
challenge='thirty days of python'
print(challenge.swapcase())

THIRTY DAYS OF PYTHON


startswith(): Checks if String Starts with the Specified String

In [32]:
challenge='thirty days of python'
print(challenge.startswith('who'))
print(challenge.startswith('thirty'))

False
True


Concatenate the string 'Thirty', 'Days', 'Of', 'Python' to a single string, 'Thirty Days Of Python'.
Concatenate the string 'Coding', 'For' , 'All' to a single string, 'Coding For All'.

In [38]:
a='Thirty'
b='Days'
c='Of'
d='Python'
space=' '
f=a + space+ b + space+ c + space+ d
print(f)

Thirty Days Of Python


In [46]:
res='Thirty' + " " + 'Days' + " " + 'Of' + " " + 'Python'
print(res)

Thirty Days Of Python


In [42]:
a="Coding"
b="For"
c="All"
space=" "
print(f"{a}{space}{b}{space}{c}")

Coding For All


In [44]:
result='Coding' + " " + 'For' + " " + 'All'
print(result)

Coding For All


Declare a variable named company and assign it to an initial value "Coding For All".
Print the variable company using print().
Print the length of the company string using len() method and print().
Change all the characters to uppercase letters using upper() method.
Change all the characters to lowercase letters using lower() method.

In [48]:
company="Coding For All"
print(company)
print(len(company))
print(company.upper())
print(company.lower())

Coding For All
14
CODING FOR ALL
coding for all


Use capitalize(), title(), swapcase() methods to format the value of the string Coding For All.

In [49]:
value='Coding For All'
print(value.capitalize())
print(value.title())
print(value.swapcase())

Coding for all
Coding For All
cODING fOR aLL


Cut(slice) out the first word of Coding For All string.

In [52]:
a='Coding For All'
print(a[0:1])
print(a[0:6])

C
Coding


Check if Coding For All string contains a word Coding using the method index, find or other methods.

In [55]:
a='Coding For All'
print(a.find(''))

-1


In [57]:
sentence='Coding For All'
sub_string='Coding'
index=sentence.find(sub_string)
print(index)

print('Coding' in 'Coding For All')

0
True


Replace the word coding in the string 'Coding For All' to Python.
Change Python for Everyone to Python for All using the replace method or other methods.

In [58]:
sentence='Coding For All'
print(sentence.replace('Coding','Python'))

Python For All


In [59]:
sentence='Python For Everyone'
print(sentence.replace('Everyone','All'))

Python For All


Split the string 'Coding For All' using space as the separator (split()) .
"Facebook, Google, Microsoft, Apple, IBM, Oracle, Amazon" split the string at the comma.

In [3]:
a='Coding For All'
print(len(a))
c=a.split(' , ')
print(c)
print(len(c))

14
['Coding For All']
1


In [9]:
string1 = 'Coding For All'
result1 = string1.split()
print(result1)


['Coding', 'For', 'All']


In [10]:
b="Facebook,Google,Microsoft,Apple,IBM,Oracle,Amazon"
c=b.split()
print(c)

['Facebook,Google,Microsoft,Apple,IBM,Oracle,Amazon']


In [7]:
string2 = "Facebook, Google, Microsoft, Apple, IBM, Oracle, Amazon"
result2 = string2.split(', ')
print(result2)


['Facebook', 'Google', 'Microsoft', 'Apple', 'IBM', 'Oracle', 'Amazon']


What is the character at index 0 in the string Coding For All.
What is the last index of the string Coding For All.
What character is at index 10 in "Coding For All" string.


In [11]:
string="coding for all"
c=string[0]
print(c)

c


In [14]:
string='Coding For All'
last_index=len(string)-1
last_letter=string[last_index]
print(last_letter)

l


In [15]:
string="coding For All"
c=string[10]
print(c)

Create an acronym or an abbreviation for the name 'Python For Everyone'.
Create an acronym or an abbreviation for the name 'Coding For All'.


Python For Everyone:

Acronym: PFE
Abbreviation: PyFE
Coding For All:

Acronym: CFA
Abbreviation: C4A

Use index to determine the position of the first occurrence of C in Coding For All.
Use index to determine the position of the first occurrence of F in Coding For All.
Use rfind to determine the position of the last occurrence of l in Coding For All People.

In [16]:
string='Coding For All'
sub_string='C'
print(string.index(sub_string))

0


In [17]:
string='Coding For All'
sub_string="F"
print(string.index(sub_string))

7


In [18]:
string="Coding For All People"
sub_string='l'
print(string.rfind(sub_string))

19


Use index or find to find the position of the first occurrence of the word 'because' in the following sentence: 'You cannot end a sentence with because because because is a conjunction'

Use rindex to find the position of the last occurrence of the word because in the following sentence: 'You cannot end a sentence with because because because is a conjunction'

Slice out the phrase 'because because because' in the following sentence: 'You cannot end a sentence with because because because is a conjunction'

Find the position of the first occurrence of the word 'because' in the following sentence: 'You cannot end a sentence with because because because is a conjunction'

In [25]:
string='You cannot end a sentence with because because because is a conjunction'
sub_string='because'
print(string.find(sub_string))
print(string.index(sub_string))
print(string.rindex(sub_string))
print(string.rfind(sub_string))
print(string[31:54])


31
31
47
47
because because because


Does ''Coding For All' start with a substring Coding?

Does 'Coding For All' end with a substring coding?

'   Coding For All      '  , remove the left and right trailing spaces in the given string.

Which one of the following variables return True when we use the method isidentifier():

30DaysOfPython
thirty_days_of_python

In [27]:
string="Coding For All"
print(string.startswith('Coding'))
print(string.endswith('coding'))

True
False


In [33]:
original_string='   Coding For All      '
trimmed_string=original_string.strip()
print(trimmed_string)



   Coding For All      


In [35]:
string='30daysofpython'
print(string.isidentifier())

False


In [39]:
a='thirty_days_of_python'
print(a.isidentifier())

True


The following list contains the names of some of python libraries: ['Django', 'Flask', 'Bottle', 'Pyramid', 'Falcon']. Join the list with a hash with space string.


In [43]:
python_libraries=['Django','Flask',"bottle",'Pyramid','Falcon']
c=' # '.join(python_libraries)
print(c)

Django # Flask # bottle # Pyramid # Falcon


Use the new line escape sequence to separate the following sentences.
I am enjoying this challenge.
I just wonder what is next.


Use a tab escape sequence to write the following lines.
Name      Age     Country   City
Asabeneh  250     Finland   Helsinki

In [45]:
print('I am enjoying this challenge.\n I just wonder what is next.')

I am enjoying this challenge.
 I just wonder what is next.


In [46]:
print('Name\tAge\tCounty\tCity')
print('Mouniks\t22\tIndia\tHyderabad')

Name	Age	County	City
Mouniks	22	India	Hyderabad


Make the following using string formatting methods:
8 + 6 = 14
8 - 6 = 2
8 * 6 = 48
8 / 6 = 1.33
8 % 6 = 2
8 // 6 = 1
8 ** 6 = 262144

In [57]:
a=8
b=6
print('{} + {} = {}'.format(a,b,a+b))
print('{} - {} = {}'.format(a,b,a-b))
print('{} * {} = {}'.format(a,b,a*b))
print('{} / {} = {:.2f}'.format(a,b,a/b))
print('{} % {} ={}'.format(a,b,a%b))
print('{} // {}={}'.format(a,b,a//b))
print('{}**{}={}'.format(a,b,a**b))

8 + 6 = 14
8 - 6 = 2
8 * 6 = 48
8 / 6 = 1.33
8 % 6 =2
8 // 6=1
8**6=262144



Use the string formatting method to display the following:
radius = 10
area = 3.14 * radius ** 2
The area of a circle with radius 10 is 314 meters squ

In [61]:
radius=10
area=3.14*radius**2
print('The area of a circle with radius {} is {} meters square'.format(str(radius),str(area)))

The area of a circle with radius 10 is 314.0 meters square


In [63]:

# Single line comment
letter = 'P'                # A string could be a single character or a bunch of texts
print(letter)               # P
print(len(letter))          # 1
greeting = 'Hello, World!'  # String could be  a single or double quote,"Hello, World!"
print(greeting)             # Hello, World!
print(len(greeting))        # 13
sentence = "I hope you are enjoying 30 days of python challenge"
print(sentence)

# Multiline String
multiline_string = '''I am a teacher and enjoy teaching.
I didn't find anything as rewarding as empowering people.
That is why I created 30 days of python.'''
print(multiline_string)
# Another way of doing the same thing
multiline_string = """I am a teacher and enjoy teaching.
I didn't find anything as rewarding as empowering people.
That is why I created 30 days of python."""
print(multiline_string)

# String Concatenation
first_name = 'Asabeneh'
last_name = 'Yetayeh'
space = ' '
full_name = first_name  +  space + last_name
print(full_name) # Asabeneh Yetayeh
# Checking length of a string using len() builtin function
print(len(first_name))  # 8
print(len(last_name))   # 7
print(len(first_name) > len(last_name)) # True
print(len(full_name)) # 15

#### Unpacking characters 
language = 'Python'
a,b,c,d,e,f = language # unpacking sequence characters into variables
print(a) # P
print(b) # y
print(c) # t 
print(d) # h
print(e) # o
print(f) # n

# Accessing characters in strings by index
language = 'Python'
first_letter = language[0]
print(first_letter) # P
second_letter = language[1]
print(second_letter) # y
last_index = len(language) - 1
last_letter = language[last_index]
print(last_letter) # n

# If we want to start from right end we can use negative indexing. -1 is the last index
language = 'Python'
last_letter = language[-1]
print(last_letter) # n
second_last = language[-2]
print(second_last) # o

# Slicing

language = 'Python'
first_three = language[0:3] # starts at zero index and up to 3 but not include 3
last_three = language[3:6]
print(last_three) # hon
# Another way
last_three = language[-3:]
print(last_three)   # hon
last_three = language[3:]
print(last_three)   # hon

# Skipping character while splitting Python strings
language = 'Python'
pto = language[0:6:2] # 
print(pto) # pto

# Escape sequence
print('I hope every one enjoying the python challenge.\nDo you ?') # line break
print('Days\tTopics\tExercises')
print('Day 1\t3\t5')
print('Day 2\t3\t5')
print('Day 3\t3\t5')
print('Day 4\t3\t5')
print('This is a back slash  symbol (\\)') # To write a back slash
print('In every programming language it starts with \"Hello, World!\"')

## String Methods
# capitalize(): Converts the first character the string to Capital Letter

challenge = 'thirty days of python'
print(challenge.capitalize()) # 'Thirty days of python'

# count(): returns occurrences of substring in string, count(substring, start=.., end=..)

challenge = 'thirty days of python'
print(challenge.count('y')) # 3
print(challenge.count('y', 7, 14)) # 1
print(challenge.count('th')) # 2`

# endswith(): Checks if a string ends with a specified ending

challenge = 'thirty days of python'
print(challenge.endswith('on'))   # True
print(challenge.endswith('tion')) # False

# expandtabs(): Replaces tab character with spaces, default tab size is 8. It takes tab size argument

challenge = 'thirty\tdays\tof\tpython'
print(challenge.expandtabs())   # 'thirty  days    of      python'
print(challenge.expandtabs(10)) # 'thirty    days      of        python'

# find(): Returns the index of first occurrence of substring

challenge = 'thirty days of python'
print(challenge.find('y'))  # 5
print(challenge.find('th')) # 0

# format()	formats string into nicer output    
first_name = 'Asabeneh'
last_name = 'Yetayeh'
job = 'teacher'
country = 'Finland'
sentence = 'I am {} {}. I am a {}. I live in {}.'.format(first_name, last_name, job, country)
print(sentence) # I am Asabeneh Yetayeh. I am a teacher. I live in Finland.

radius = 10
pi = 3.14
area = pi # radius ## 2
result = 'The area of circle with {} is {}'.format(str(radius), str(area))
print(result) # The area of circle with 10 is 314.0

# index(): Returns the index of substring
challenge = 'thirty days of python'
print(challenge.find('y'))  # 5
print(challenge.find('th')) # 0

# isalnum(): Checks alphanumeric character

challenge = 'ThirtyDaysPython'
print(challenge.isalnum()) # True

challenge = '30DaysPython'
print(challenge.isalnum()) # True

challenge = 'thirty days of python'
print(challenge.isalnum()) # False

challenge = 'thirty days of python 2019'
print(challenge.isalnum()) # False

# isalpha(): Checks if all characters are alphabets

challenge = 'thirty days of python'
print(challenge.isalpha()) # True
num = '123'
print(num.isalpha())      # False

# isdecimal(): Checks Decimal Characters

challenge = 'thirty days of python'
print(challenge.find('y'))  # 5
print(challenge.find('th')) # 0

# isdigit(): Checks Digit Characters

challenge = 'Thirty'
print(challenge.isdigit()) # False
challenge = '30'
print(challenge.isdigit())   # True

# isdecimal():Checks decimal characters

num = '10'
print(num.isdecimal()) # True
num = '10.5'
print(num.isdecimal()) # False


# isidentifier():Checks for valid identifier means it check if a string is a valid variable name

challenge = '30DaysOfPython'
print(challenge.isidentifier()) # False, because it starts with a number
challenge = 'thirty_days_of_python'
print(challenge.isidentifier()) # True


# islower():Checks if all alphabets in a string are lowercase

challenge = 'thirty days of python'
print(challenge.islower()) # True
challenge = 'Thirty days of python'
print(challenge.islower()) # False

# isupper(): returns if all characters are uppercase characters

challenge = 'thirty days of python'
print(challenge.isupper()) #  False
challenge = 'THIRTY DAYS OF PYTHON'
print(challenge.isupper()) # True


# isnumeric():Checks numeric characters

num = '10'
print(num.isnumeric())      # True
print('ten'.isnumeric())    # False

# join(): Returns a concatenated string

web_tech = ['HTML', 'CSS', 'JavaScript', 'React']
result = '#, '.join(web_tech)
print(result) # 'HTML# CSS# JavaScript# React'

# strip(): Removes both leading and trailing characters

challenge = ' thirty days of python '
print(challenge.strip('y')) # 5

# replace(): Replaces substring inside

challenge = 'thirty days of python'
print(challenge.replace('python', 'coding')) # 'thirty days of coding'

# split():Splits String from Left

challenge = 'thirty days of python'
print(challenge.split()) # ['thirty', 'days', 'of', 'python']

# title(): Returns a Title Cased String

challenge = 'thirty days of python'
print(challenge.title()) # Thirty Days Of Python

# swapcase(): Checks if String Starts with the Specified String
  
challenge = 'thirty days of python'
print(challenge.swapcase())   # THIRTY DAYS OF PYTHON
challenge = 'Thirty Days Of Python'
print(challenge.swapcase())  # tHIRTY dAYS oF pYTHON

# startswith(): Checks if String Starts with the Specified String

challenge = 'thirty days of python'
print(challenge.startswith('thirty')) # True
challenge = '30 days of python'
print(challenge.startswith('thirty')) # False

P
1
Hello, World!
13
I hope you are enjoying 30 days of python challenge
I am a teacher and enjoy teaching.
I didn't find anything as rewarding as empowering people.
That is why I created 30 days of python.
I am a teacher and enjoy teaching.
I didn't find anything as rewarding as empowering people.
That is why I created 30 days of python.
Asabeneh Yetayeh
8
7
True
16
P
y
t
h
o
n
P
y
n
n
o
hon
hon
hon
Pto
I hope every one enjoying the python challenge.
Do you ?
Days	Topics	Exercises
Day 1	3	5
Day 2	3	5
Day 3	3	5
Day 4	3	5
This is a back slash  symbol (\)
In every programming language it starts with "Hello, World!"
Thirty days of python
3
1
2
True
False
thirty  days    of      python
thirty    days      of        python
5
0
I am Asabeneh Yetayeh. I am a teacher. I live in Finland.
The area of circle with 10 is 3.14
5
0
True
True
False
False
False
False
5
0
False
True
True
False
False
True
True
False
False
True
True
False
HTML#, CSS#, JavaScript#, React
 thirty days of python 
thirty days